besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -894,11 +908,25 @@ private BesuCommand startPlugins() {
             besuController.getProtocolManager().getBlockBroadcaster(),
             besuController.getTransactionPool(),
             besuController.getSyncState()));
-    besuPluginContext.addService(MetricsSystem.class, getMetricsSystem());
+    addPluginMetrics(besuController);
     besuPluginContext.startPlugins();
     return this;
   }
 
+  private void addPluginMetrics(final BesuController<?> besuController) {
+    besuPluginContext.addService(MetricsSystem.class, getMetricsSystem());
+
+    Object consensusState = besuController.getProtocolContext().getConsensusState();
+
+    if (consensusState != null && PoAContext.class.isAssignableFrom(consensusState.getClass())) {
+      PoAMetricServiceImpl service =
+          new PoAMetricServiceImpl(
+              ((IbftContext) consensusState).getBlockInterface(),
```

Comment:
Don't you want to cast to a `PoAContext` here?

Reply:
Yes!